datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -654,9 +654,12 @@ impl Stream for GroupedHashAggregateStream {
                             }
 
                             if let Some(to_emit) = self.group_ordering.emit_to() {
-                                let batch = extract_ok!(self.emit(to_emit, false));
-                                self.exec_state = ExecutionState::ProducingOutput(batch);
                                 timer.done();
+                                let Some(batch) = extract_ok!(self.emit(to_emit, false))
+                                else {
+                                    break 'reading_input;
+                                };
+                                self.exec_state = ExecutionState::ProducingOutput(batch);
```

Comment:
Nit: I think I would find this easier to read if it it avoid a redundant `break`:

```suggestion
                                if let let Some(batch) = extract_ok!(self.emit(to_emit, false)) {
                                  self.exec_state = ExecutionState::ProducingOutput(batch);
                                }
```